### ISSUE TITLE
If Platform.environment. containsKey('CI') – update PUB_ENVIRONMENT

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/pub#2758]
We want to know if pub invocations are coming from CI

Right now, we have to hard-wire configurations in each CI provider to set something.

Instead, the pub client should just sniff this environment var and use it.

See

https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables
https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

| Provided PUB_ENVIRONMENT | Effective PUB_ENVIRONMENT |
|--------------------------|---------------------------|
| _Not set or empty_       | `CI_set`                  |
| `.some_value`            | `CI_set.some_value`       |
| `some_value`             | `CI_set.some_value`       |

Basically, the `CI_set` bit should always be first, and it should always end with `.` if there is a value after it – but don't put in two `.` if ones already there!

**Comment by kevmoo** [comment]
@jonasfj – can this be ditched? We just stopped doing any telemetry if `CI` exists, right?

**Comment by jonasfj** [comment]
Yes, let's close this:
https://github.com/dart-lang/pub/blob/4b5450d396a0388bf80a2990e5d21f0d6d358883/lib/src/http.dart#L88-L91
